apps/minifront/src/state/send/index.ts

```diff
@@ -111,7 +119,42 @@ export const createSendSlice = (): SliceCreator<SendSlice> => (set, get) => {
   };
 };
 
-const assembleRequest = ({ amount, feeTier, recipient, selection, memo }: SendSlice) => {
+const assembleRequest = ({
+  amount,
+  feeTier,
+  recipient,
+  selection,
+  memo,
+  isSendingMax,
+}: SendSlice) => {
+  if (isSendingMax) {
+    return new TransactionPlannerRequest({
+      spends: [
+        {
+          address: { altBech32m: recipient },
+          value: {
+            amount: toBaseUnit(
+              BigNumber(amount),
+              getDisplayDenomExponentFromValueView.optional()(selection?.balanceView),
+            ),
+            assetId: getAssetIdFromValueView(selection?.balanceView),
+          },
+        },
+      ],
+      source: getAddressIndex(selection?.accountAddress),
+
+      // Note: we currently don't provide a UI for setting the fee manually. Thus,
+      // a `feeMode` of `manualFee` is not supported here.
+      feeMode:
+        typeof feeTier === 'undefined'
+          ? { case: undefined }
+          : {
+              case: 'autoFee',
+              value: { feeTier },
+            },
+    });
+  }
+
```

Comment:
Looks like the `spends` vs. `outputs` key is the only thing that's different in the two cases code below. Could we DRY up the code here a bit? e.g., 

```suggestion
  const spendOrOutput:
    | PartialMessage<TransactionPlannerRequest_Spend>
    | PartialMessage<TransactionPlannerRequest_Output> = {
    address: { altBech32m: recipient },
    value: {
      amount: toBaseUnit(
        BigNumber(amount),
        getDisplayDenomExponentFromValueView.optional()(selection?.balanceView),
      ),
      assetId: getAssetIdFromValueView(selection?.balanceView),
    },
  };

  return new TransactionPlannerRequest({
    ...(isSendingMax ? { spends: [spendOrOutput] } : { outputs: [spendOrOutput] }),
    source: getAddressIndex(selection?.accountAddress),

    // Note: we currently don't provide a UI for setting the fee manually. Thus,
    // a `feeMode` of `manualFee` is not supported here.
    feeMode:
      typeof feeTier === 'undefined'
        ? { case: undefined }
        : {
            case: 'autoFee',
            value: { feeTier },
          },

    memo: new MemoPlaintext({
      returnAddress: getAddress(selection?.accountAddress),
      text: memo,
    }),
  });

```

Reply:
good candidate for DRY, changed